### ISSUE TITLE
Scratchpad

### ISSUE THREAD

**Issue by ymoisan** [NRCan/geo-deep-learning#60]
- log files in JSON (easier parsing and future interop ?)  
- Trap standard error (AWS) and add full python traceback in output.txt
- Fonctions lambda dans GH ?? (t3.large revenu!)  
- possible memory leak during images_to_samples.py  (`with` file ...; wrap in a Python lambda ?)  
- Log files renaming --> add the config file name in the log name

**Comment by mpelchat04** [comment]
* Log files renaming --> add the config file name in the log name

**Comment by mpelchat04** [comment]
# Active Learning  
[Active learning](http://www.cs.cmu.edu/~bsettles/pub/settles.activelearning.pdf) aims to facilitate the annotation process by letting the model choose candidates for new annotations. After a phase of training, the model runs inference on a stack of unlabeled images. We then calculate [entropy mesure](https://towardsdatascience.com/entropy-is-a-measure-of-uncertainty-e2c000301c2c) on the predicted sample and use samples with low entropy (or uncertainty) as new candidate for annotations. The annotator is put to contribution in order to accept/reject the proposed annotations. The accepted annotations are then incorporated into the training dataset and we start a new iteration of training.  
How the annotations will be accepted/rejected is still to be determined...

**Comment by mpelchat04** [comment]
# Test the Intel Python distribution  
Intel claims their [Python distro](https://software.intel.com/content/www/us/en/develop/tools/distribution-for-python.html) is much faster for anything numeric. Since it can be installed with conda, can we give it a try ?

**Comment by mpelchat04** [comment]
# Federated learning  
"In a typical machine learning system, an optimization algorithm like Stochastic Gradient Descent (SGD) runs on a large dataset partitioned homogeneously across servers in the cloud. Such highly iterative algorithms require low-latency, high-throughput connections to the training data. But in the Federated Learning setting, the data is distributed across millions of devices in a highly uneven fashion. In addition, these devices have significantly higher-latency, lower-throughput connections and are only intermittently available for training."
https://ai.googleblog.com/2017/04/federated-learning-collaborative.html

Could this be applied in a later stage, e.g. to optimize our model through time ?
